docs/run/theta.rst

```diff
@@ -48,17 +48,59 @@ Print jobs referenced in Balsam database::
 
         export PROJECT_NAME=datascience
 
-Finally you can submit a cobalt job to Theta which will start by running
+Finally, you can submit a Cobalt job to Theta which will start by running
 your master job named test::
 
     balsam submit-launch -n 128 -q default -t 180 -A $PROJECT_NAME --job-mode serial --wf-filter TEST
 
-
-Now if you want to look at the logs, go to ``testdb/data/TEST``. You'll see
+If you want to look at the logs, navigate to ``testdb/data/TEST``. You'll see
 one directory prefixed with ``test``. Inside this directory you will find the
-logs of you search. All the other directories prefixed with ``task`` correspond
-to the logs of your ``--run`` function, here the run function is corresponding
-to the training of a neural network.
+logs of you search. All the other directories prefixed with ``task`` contain the
+individual logs of each evaluation your ``--run`` function. Here, the run function 
+corresponds to the training of a neural network.
+
+The above specification of Balsam ``--job-mode=serial`` can be used for tasks that **do
+not** use MPI (and hence are restricted to a single node). With the current settings,
+Balsam executes a ``Master`` orchestrator process on the entirety of the first node, and
+each of the remaining 127 nodes will execute isolated DeepHyper "workers". One worker
+(compute node) will execute the AMBS searcher/learner process. The remaining 126 compute
+nodes will execute independent evaluators defined by
+``deephyper.benchmark.hps.polynome2.run``. Only a single call to the job scheduler's
+``mpirun``-equivalent command is made at the beginning of ``submit-launch`` command; jobs
+are run as forked processes on all of the ``Worker`` ranks.
+
+DeepHyper automatically infers the number of available workers from the Balsam environment
+variables ``BALSAM_LAUNCHER_NODES`` and ``BALSAM_JOB_MODE``. Additionally, the user can
+define the ``DEEPHYPER_WORKERS_PER_NODE`` environment variable to pack several tasks per
+node in this Balsam job mode. In that case, the ``--node-packing-count`` option of
+``balsam job`` should be changed from its default value of 1 to be consistent with the
+environment variable and ensure full job occupancy of the ``Worker`` ranks. For example,
+to run 4 DeepHyper workers per node, the above command is modified::
+
+  balsam job --name test --application AMBS --workflow TEST --node-packing-count=4 --env DEEPHYPER_WORKERS_PER_NODE=4 --args '--evaluator balsam --problem deephyper.benchmark.hps.polynome2.Problem --run deephyper.benchmark.hps.polynome2.run'
```

Comment:
To clarify:  the ``--node-packing-count`` ensures the AMBS search itself only takes a fraction of the node.  The ``--env DEEPHYPER_WORKERS_PER_NODE=4`` takes care of ensuring that each evaluation ``task`` also has the same fractional occupancy.   These kind of dependent parameters should be taken care of by the DeepHyper CLI (there is some work along these lines in the develop branch)

Reply:
yeah, it took me a bit to realize that the subsequent spawning of each evaluation `task` used the environment variable, but it makes sense. 

Perhaps my example isnt the most illustrative, since I set the values to be equal (but I imagine that it is the most common use-case).  Might be good to add another example with different occupancies, and `--n-jobs>1` for the "learner".

Reply:
Added in cfb8d0fc236290e1e0c60f2b587b870dc29e5a54